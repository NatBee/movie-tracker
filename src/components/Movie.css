* {
  margin: 0;
  padding: 0; }

.movie {
  width: 350px;
  margin: 15px;
  height: 500px; }

.background {
  height: 500px;
  background-size: cover; }

.description {
  position: relative;
  top: -500px;
  visibility: hidden;
  /*display: none;*/ }

.full:hover .background {
  filter: blur(5px); }

.full:hover .description {
  color: red;
  visibility: visible;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 450px;
  width: 300px;
  background-color: black;
  opacity: 0.7;
  color: white;
  padding: 25px; }

/*visibi*/
/*lity: visible;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    height: 450px;
    width: 300px;
    background-color: black;
    opacity: 0.7;
    color: white;
    padding: 25px;*/
