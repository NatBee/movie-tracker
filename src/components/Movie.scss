* {
  margin: 0;
  padding: 0;
}

.movie {
  height: 450px;
  margin: 15px;
}

.background {
  background-size: cover;
  height: 450px;
  width: 300px;
}

.description {
  display: flex;
  background-color: black;
  color: white;
  flex-direction: column;
  height: 400px;
  justify-content: space-around;
  opacity: 0.7;
  padding: 25px;
  position: relative;
  top: -450px;
  visibility: hidden;
  width: 250px;
}

.full:hover {
  .background {
    filter: blur(5px);
  }

  .description {
    visibility: visible;
  }
}