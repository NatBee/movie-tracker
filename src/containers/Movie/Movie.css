* {
  margin: 0;
  padding: 0;
  font-family: roboto; }

.movie {
  height: 450px;
  margin: 15px; }

.background {
  background-size: cover;
  height: 450px;
  width: 300px; }

.description {
  display: flex;
  background-color: black;
  color: white;
  flex-direction: column;
  height: 400px;
  justify-content: space-around;
  opacity: 0.7;
  padding: 25px;
  position: relative;
  top: -450px;
  visibility: hidden;
  width: 250px; }

.favorite-icon {
  width: 40px;
  height: 40px; }

.full:hover .background {
  filter: blur(5px); }

.full:hover .description {
  visibility: visible; }
