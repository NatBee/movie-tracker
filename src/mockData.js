export const mockData = [{
  "vote_count": 20,
  "id": 337167,
  "video": false,
  "vote_average": 6.4,
  "title": "Fifty Shades Freed",
  "popularity": 45.670468,
  "poster_path": "/iUDFe6B6bFPOAE13uoqH4eEU1cF.jpg",
  "original_language": "en",
  "original_title": "Fifty Shades Freed",
  "genre_ids": [18, 10749],
  "backdrop_path": "/iuyE1qy9HoTsMg0DHB1KJxM94DJ.jpg",
  "adult": false,
  "overview": "The third installment of the Fifty Shades Trilogy.",
  "release_date": "2018-02-07"
}];