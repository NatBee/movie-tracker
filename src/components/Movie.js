import React, {Component} from 'react';
import { Link, redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { toggleFavorite } from '../actions/index';
import { addFavorite } from '../apiCall.js';
import icon from '../like.png';
import './Movie.css';

class Movie extends Component {
  renderMovie = () => {
    const posterPath = this.props.poster; 
    const url = `https://image.tmdb.org/t/p/w500${ posterPath }`
    const backgroundImage = {
      backgroundImage: `url(${ url })`
    }

    return (
      <div className="full">
        <div className="background" style={ backgroundImage }>
        </div>
        <div className="description" >
          <h3>Title: { this.props.title }</h3>
          <p>Release Date: { this.props.releaseDate }</p>
          <p>Popularity: { this.props.rating } %</p>
          <p>Overview: { this.props.overview }</p>
          <img src={ icon }
               alt="favorite button" 
               className="favorite-icon" 
               onClick={this.favoriteHandler} />
        </div>
      </div>
    )
  }

  favoriteHandler = () => {

    if(this.props.current_user.id) {
      const userId = this.props.current_user.id;
      const movie = this.props;

      addFavorite(userId, movie);
    } else {
      this.props.history.push('/login');
    }
  }

  handleFavorite = () => {   
    this.props.toggleFavorite(this.props.id);
    this.addToFavoritesArray();
  }

  addToFavoritesArray = () => {
    const movies = this.props.movies;
    const favorites = movies.filter( movie => movie.favorite === true);
  }

  // for displaying individual movies:
  // const displayMovies = movies.map((movie, i) => {
  //   return (
  //     <Link to={`/movies/${movie.id}`} >
  //       <img src={movie.image} key={i} />
  //     </Link>
  //   )
  // })


  render() {
    return (
      <div className="movie">
        { this.renderMovie() }
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  movies: store.movies, 
  favorite: store.favorite,
  current_user: store.current_user,
})

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: movieId => dispatch(toggleFavorite(movieId))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Movie));