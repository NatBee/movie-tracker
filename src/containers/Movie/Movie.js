import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { addFavorite, getFavorites, deleteFavorites } from '../../helper/apiCall.js';
import iconInactive from '../../assets/like.png';
import iconActive from '../../assets/like-active.png';
import './Movie.css';
import PropTypes from 'prop-types';

export class Movie extends Component {
  constructor(props) {
    super(props);

    this.state = {
      favorite: false
    }
  }

  componentDidMount() {
    this.setFavoriteState()
  }

  setFavoriteState = async () => {
    const userId = this.props.current_user.id
    const match = await this.findMatch(userId, this.props)

    if (match) {
      this.setState({ favorite: true })
    } else {
      this.setState({ favorite: false })
    }
  }

  findMatch = async (userId, movie) => {
    const { data } = await getFavorites(userId);
    const match = data.find( favorite => {
      return (movie.id === favorite.movie_id) ||
             (movie.id === favorite.id)
    })

    return match
  }

  checkFavorite = async () => {
    const userId = this.props.current_user.id
    const match = await this.findMatch(userId, this.props)

    if (match) {
      const movieId = match.movie_id;

      this.setState({ favorite: false })
      return deleteFavorites(userId, movieId)
    } else {
      this.setState({ favorite: true })
      return addFavorite(userId, this.props)
    }
  }

  favoriteHandler = () => {
    if(this.props.current_user.id) {
      this.checkFavorite();
    } else {
      this.props.history.push('/login');
    }
  }

  setIcon = () => {
    return this.state.favorite ? iconActive : iconInactive
  }

  renderMovie = () => {
    const posterPath = this.props.poster; 
    const url = `https://image.tmdb.org/t/p/w500${ posterPath }`
    const backgroundImage = {
      backgroundImage: `url(${ url })`
    }

    return (
      <div className="full">
        <div className="background" style={ backgroundImage }>
        </div>
        <div className="description" >
          <h3>Title: { this.props.title }</h3>
          <p>Release Date: { this.props.releaseDate }</p>
          <p>Popularity: { this.props.rating } %</p>
          <p>Overview: { this.props.overview }</p>
          <img src={ this.setIcon() }
               alt="favorite button" 
               className="favorite-icon" 
               onClick={ this.favoriteHandler } />
        </div>
      </div>
    )
  }

  render() {
    return (
      <div className="movie">
        { this.renderMovie() }
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  movies: store.movies, 
  current_user: store.current_user,
})

Movie.propTypes = {
  history: PropTypes.object,
  setIcon: PropTypes.func,
  renderMovie: PropTypes.func,
  checkFavorite: PropTypes.func,
  favoriteHandler: PropTypes.func,
  movies: PropTypes.array,
  current_user: PropTypes.object
}

export default withRouter(connect(mapStateToProps, null)(Movie));