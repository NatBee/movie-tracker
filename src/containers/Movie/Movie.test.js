/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import { Movie, mapStateToProps, mapDispatchToProps } from './Movie';
import mockData from '../../mockData';

describe('Movie', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Movie />);


  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should map to the store correctly', () => {
    const mockCurrentUser = {
      "id": 1,
      "name": "Taylor",
      "password": "password",
      "email": "tman2272@aol.com"
    }
    const mockMovie = mockData;
    const mockStore = {
      current_user: mockCurrentUser,
      movies: mockMovie
    }
    const mapped = mapStateToProps(mockStore);

    expect(mapped).toEqual(mockStore)
  })

  it('should call checkFavorite when the favorite button is selected and favoriteHandler is called', () => {
    const mockFavoriteHandler = jest.fn();
    const mockCheckFavorite = jest.fn();
    const mockCurrentUser = {
      "id": 1,
      "name": "Taylor",
      "password": "password",
      "email": "tman2272@aol.com"
    }
    const wrapper = shallow(<Movie favoriteHandler={mockFavoriteHandler} 
                                   checkFavorite={mockCheckFavorite} 
                                   current_user={mockCurrentUser}

                                   /> );

    wrapper.instance().favoriteHandler();

    expect(mockFavoriteHandler).toHaveBeenCalled();
  })


})