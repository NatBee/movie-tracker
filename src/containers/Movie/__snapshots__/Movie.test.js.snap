// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="full"
      >
        <div
          className="background"
          style={
            Object {
              "backgroundImage": "url(https://image.tmdb.org/t/p/w500undefined)",
            }
          }
        />
        <div
          className="description"
        >
          <h3>
            Title: 
          </h3>
          <p>
            Release Date: 
          </p>
          <p>
            Popularity: 
             %
          </p>
          <p>
            Overview: 
          </p>
          <img
            alt="favorite button"
            className="favorite-icon"
            onClick={[Function]}
            src="like.png"
          />
        </div>
      </div>,
      "className": "movie",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="background"
            style={
              Object {
                "backgroundImage": "url(https://image.tmdb.org/t/p/w500undefined)",
              }
            }
          />,
          <div
            className="description"
          >
            <h3>
              Title: 
            </h3>
            <p>
              Release Date: 
            </p>
            <p>
              Popularity: 
               %
            </p>
            <p>
              Overview: 
            </p>
            <img
              alt="favorite button"
              className="favorite-icon"
              onClick={[Function]}
              src="like.png"
            />
          </div>,
        ],
        "className": "full",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "background",
            "style": Object {
              "backgroundImage": "url(https://image.tmdb.org/t/p/w500undefined)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Title: 
              </h3>,
              <p>
                Release Date: 
              </p>,
              <p>
                Popularity: 
                 %
              </p>,
              <p>
                Overview: 
              </p>,
              <img
                alt="favorite button"
                className="favorite-icon"
                onClick={[Function]}
                src="like.png"
              />,
            ],
            "className": "description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Title: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Title: ",
                undefined,
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Release Date: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Release Date: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Popularity: ",
                  undefined,
                  " %",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Popularity: ",
                undefined,
                " %",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Overview: ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Overview: ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "favorite button",
                "className": "favorite-icon",
                "onClick": [Function],
                "src": "like.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="full"
        >
          <div
            className="background"
            style={
              Object {
                "backgroundImage": "url(https://image.tmdb.org/t/p/w500undefined)",
              }
            }
          />
          <div
            className="description"
          >
            <h3>
              Title: 
            </h3>
            <p>
              Release Date: 
            </p>
            <p>
              Popularity: 
               %
            </p>
            <p>
              Overview: 
            </p>
            <img
              alt="favorite button"
              className="favorite-icon"
              onClick={[Function]}
              src="like.png"
            />
          </div>
        </div>,
        "className": "movie",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="background"
              style={
                Object {
                  "backgroundImage": "url(https://image.tmdb.org/t/p/w500undefined)",
                }
              }
            />,
            <div
              className="description"
            >
              <h3>
                Title: 
              </h3>
              <p>
                Release Date: 
              </p>
              <p>
                Popularity: 
                 %
              </p>
              <p>
                Overview: 
              </p>
              <img
                alt="favorite button"
                className="favorite-icon"
                onClick={[Function]}
                src="like.png"
              />
            </div>,
          ],
          "className": "full",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "background",
              "style": Object {
                "backgroundImage": "url(https://image.tmdb.org/t/p/w500undefined)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Title: 
                </h3>,
                <p>
                  Release Date: 
                </p>,
                <p>
                  Popularity: 
                   %
                </p>,
                <p>
                  Overview: 
                </p>,
                <img
                  alt="favorite button"
                  className="favorite-icon"
                  onClick={[Function]}
                  src="like.png"
                />,
              ],
              "className": "description",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Title: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Title: ",
                  undefined,
                ],
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Release Date: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Release Date: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Popularity: ",
                    undefined,
                    " %",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Popularity: ",
                  undefined,
                  " %",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Overview: ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Overview: ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "favorite button",
                  "className": "favorite-icon",
                  "onClick": [Function],
                  "src": "like.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
